src/components/cdTab.js

```diff
@@ -1,34 +1,183 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-import Grid from '@mui/material/Grid'
-import { Skeleton } from '@mui/material'
+import {
+  Alert,
+  Box,
+  FormControlLabel,
+  FormGroup,
+  Switch,
+  Tooltip,
+} from '@mui/material'
 
 import ObjectTextViewer from './util/ObjectTextViewer'
+import { useFetchComponentDescriptor } from '../api/useFetch'
+import CenteredSpinner from './util/CenteredSpinner'
+
+
+const FetchRawComponent = ({
+  setRaw,
+  setLoading,
+  setIsError,
+  componentName,
+  componentVersion,
+  ocmRepo,
+  versionFilter,
+}) => {
+  // eslint-disable-next-line no-unused-vars
+  const [componentDescriptor, isLoading, isError, _] = useFetchComponentDescriptor({
+    componentName: componentName,
+    ocmRepoUrl: ocmRepo,
+    version: componentVersion,
+    versionFilter: versionFilter,
+    raw: true,
+  })
+
+  React.useEffect(() => {
+    setRaw(componentDescriptor)
+    setLoading(isLoading)
+    setIsError(isError)
+  }, [
+    componentDescriptor,
+    isLoading,
+    isError,
+    setRaw,
+    setLoading,
+    setIsError,
+  ])
+}
+FetchRawComponent.displayName = 'FetchRawComponent'
+FetchRawComponent.propTypes = {
+  setRaw: PropTypes.func.isRequired,
+  setLoading: PropTypes.func.isRequired,
+  setIsError: PropTypes.func.isRequired,
+  componentName: PropTypes.string.isRequired,
+  componentVersion: PropTypes.string.isRequired,
+  ocmRepo: PropTypes.string,
+  versionFilter: PropTypes.string,
+}
+
+
+const Options = ({
+  isLoading,
+  raw,
```

Comment:
Let us not mixup `raw` with `rawMode` here, in that, rather rename these parameters to `rawMode` and `setRawMode`